docs/content/guides/developer/app-examples/e2e-counter.mdx

```diff
@@ -179,124 +185,196 @@ The UI design consists of two parts:
+  ```
+</details>
+
+<details>
+  <summary>
+    Update the `src/networkConfig.ts` file to include the `packageID` constants.
+  </summary>
+  {@inject: sdk/create-dapp/templates/react-e2e-counter/src/networkConfig.ts noComments}
+</details>
 
 ### Creating `Counter`
 
-You need a way to create a new `Counter` object. Do this by creating a new `CreateCounter` component in the new `src/CreateCounter.tsx` file:
-
-```tsx
-import { Button, Container } from '@radix-ui/themes';
-
-export function CreateCounter( { onCreated }: { onCreated: (id: string) => void; } ) {
-
-  function create() {
-    // TODO
-  }
-
-  return (
-    <Container>
-      <Button
-        size="3"
-        onClick={() => {
-          create();
-        }}
-      >
-        Create Counter
-      </Button>
-    </Container>
-  );
-}
-```
+You need a way to create a new `Counter` object. 
+
+<details>
+  <summary>
+    Create `src/CreateCounter.tsx` and add the following code:
+  </summary>
+  ```tsx
+    import { Button, Container } from "@radix-ui/themes";
+    import { useState } from "react";
+    import ClipLoader from "react-spinners/ClipLoader";
+
+    export function CreateCounter({
+      onCreated,
+    }: {
+      onCreated: (id: string) => void;
+    }) {
+      const [waitingForTxn, setWaitingForTxn] = useState(false);
+
+      function create() {
+        // TODO
+      }
+
+      return (
+        <Container>
+          <Button
+            size="3"
+            onClick={() => {
+              create();
+            }}
+            disabled={waitingForTxn}
+          >
+            {waitingForTxn ? <ClipLoader size={20} /> : "Create Counter"}
+          </Button>
+        </Container>
+      );
+    }
+  ```
+</details>
 
-This component renders a button that enables the user to create a counter. Now, update your `create` function so that it calls the `create` function in your Move module.
+This component renders a button that enables the user to create a counter. Now, update your `create` function so that it calls the `create` function from your Move module.
 
-To do this, you need to construct a `Transaction`, with the appropriate `moveCall` transaction, and then sign and execute the programmable transaction block (PTB).
+<details>
+  <summary>
+    Update the `create` function in the `src/CreateCounter.tsx` file:
+  </summary>
+  {@inject: sdk/create-dapp/templates/react-e2e-counter/src/CreateCounter.tsx noComments}
+</details>
 
-First, import `Transaction` from `@mysten/sui`, `COUNTER_PACKAGE_ID` from your constants.ts file created previously, and `useSignAndExecuteTransaction` from `@mysten/dapp-kit`.
+The `create` function now creates a new Sui `Transaction` and calls the `create` function from your Move module. The Programmable Transaction Block (PTB) is then signed and executed via the `useSignAndExecuteTransaction` hook. The `onCreated` callback is called with the new counter's ID when the transaction is successful.
```

Comment:
```suggestion
The `create` function now creates a new Sui `Transaction` and calls the `create` function from your Move module. The PTB is then signed and executed via the `useSignAndExecuteTransaction` hook. The `onCreated` callback is called with the new counter's ID when the transaction is successful.
```
After you define an acro (PTB here), always use the acro from then on in the same doc.